awswrangler/lakeformation/_read.py

```diff
@@ -0,0 +1,192 @@
+        Cannot be specified alongside transaction_id
+    catalog_id : str, optional
+        The ID of the Data Catalog from which to retrieve Databases.
+        If none is provided, the AWS account ID is used by default.
+    chunked : bool, optional
+        If `True`, Wrangler returns an Iterable of DataFrames with no guarantee of chunksize.
+    use_threads : bool
+        True to enable concurrent requests, False to disable multiple threads.
+        When enabled, os.cpu_count() is used as the max number of threads.
+    boto3_session : boto3.Session(), optional
+        Boto3 Session. The default boto3 session is used if boto3_session receives None.
+    params: Dict[str, any], optional
+        Dict of parameters used to format the partiQL query. Only named parameters are supported.
+        The dict must contain the information in the form {"name": "value"} and the SQL query must contain
+        `:name`.
+
+    Returns
+    -------
+    Union[pd.DataFrame, Iterator[pd.DataFrame]]
+        Pandas DataFrame or Generator of Pandas DataFrames if chunked is passed.
+
+    Examples
+    --------
+    >>> import awswrangler as wr
+    >>> df = wr.lakeformation.read_sql_query(
+    ...     sql="SELECT * FROM my_table;",
+    ...     database="my_db",
+    ...     catalog_id="111111111111"
+    ... )
+
+    >>> import awswrangler as wr
+    >>> df = wr.lakeformation.read_sql_query(
+    ...     sql="SELECT * FROM my_table LIMIT 10;",
+    ...     database="my_db",
+    ...     transaction_id="1b62811fa3e02c4e5fdbaa642b752030379c4a8a70da1f8732ce6ccca47afdc9"
+    ... )
+
+    >>> import awswrangler as wr
+    >>> df = wr.lakeformation.read_sql_query(
+    ...     sql="SELECT * FROM my_table WHERE name=:name;",
+    ...     database="my_db",
+    ...     query_as_of_time="1611142914",
+    ...     params={"name": "'filtered_name'"}
+    ... )
+
+    """
+    if transaction_id is not None and query_as_of_time is not None:
+        raise exceptions.InvalidArgumentCombination(
+            "Please pass only one of `transaction_id` or `query_as_of_time`, not both"
+        )
+    session: boto3.Session = _utils.ensure_session(session=boto3_session)
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=session)
+    if params is None:
+        params = {}
+    for key, value in params.items():
+        sql = sql.replace(f":{key}", str(value))
+
+    args: Dict[str, Optional[str]] = {"DatabaseName": database, "Statement": sql}
+    if catalog_id:
+        args["CatalogId"] = catalog_id
+    if query_as_of_time:
+        args["QueryAsOfTime"] = query_as_of_time
+    elif transaction_id:
+        args["TransactionId"] = transaction_id
+    else:
+        _logger.debug("Neither `transaction_id` nor `query_as_of_time` were specified, beginning transaction")
+        transaction_id = client_lakeformation.begin_transaction(ReadOnly=True)["TransactionId"]
```

Comment:
Shouldn't we conclude this transaction somewhere latter with `commit` or `rollback`?

Reply:
My understanding from the docs is that we don't need to commit it since it's a Read-only transaction

Docs:
• ReadOnly – Boolean.
Indicates that that this transaction should be read only. Writes made using a read-only transaction ID
will be rejected. Read-only transactions do not need to be committed.

Reply:
I see. Makes sense indeed.